app/models/piece.rb

```diff
@@ -44,6 +44,7 @@ def valid_move?
     if self.on_the_board? false
       flash[:danger] = "Move cannot be completed."
     end
+    return false unless legal_move?(x, y)
   end
 

```

Comment:
This is great, but I am just a bit concerned that unless legal_move? may allow a move off the board.